compiler/src/iree/compiler/Dialect/Flow/Transforms/VerifyInputLegality.cpp

```diff
@@ -17,12 +19,34 @@ namespace iree_compiler {
 namespace IREE {
 namespace Flow {
 
+static llvm::cl::opt<bool>
+    clVerifyNoF64("iree-flow-verify-no-f64",
+                  llvm::cl::desc("Verify that there are no ops that use f64."),
+                  llvm::cl::init(true));
+
 namespace {
+
 class VerifyInputLegalityPass
     : public VerifyInputLegalityBase<VerifyInputLegalityPass> {
   void runOnOperation() override {
     ConversionTarget target(getContext());
-    target.markUnknownOpDynamicallyLegal([](Operation *) { return true; });
+    target.markUnknownOpDynamicallyLegal([](Operation *op) {
+      bool shouldVerifyNoF64 =
+          clVerifyNoF64.getNumOccurrences()
+              ? clVerifyNoF64
+              : !GlobalOptimizationOptions::FromFlags::get().demoteF64ToF32;
+      if (!shouldVerifyNoF64)
+        return true;
+      bool usesFloat64 = llvm::any_of(
```

Comment:
Lets add a TODO here that we should really try and query the targets this will execute on. As f64 support is steadily improving, this warning will become less relevant and could result in folks thinking this is a failure while there is no issue. Fine given the current state.

Reply:
Done